internal/server/handler_v2.go

```diff
@@ -93,65 +91,69 @@ func (s *Server) V2Resolve(
 func (s *Server) V2Node(
 	ctx context.Context, in *pbv2.NodeRequest,
 ) (*pbv2.NodeResponse, error) {
-	arcs, err := v2.ParseProperty(in.GetProperty())
+	if in.GetNextToken() == "" {
+		// When request |next_token| is empty, there are two cases:
+		// 1. The call does not need pagination, e.g. PropertyLabels.
+		// 2. The call needs pagination, but this is the first call/page.
+		// In both cases, we need to read from both local and remote, and merge.
+
+		localResp, err := V2NodeCore(ctx, s.store, s.cache, s.metadata, in)
+		if err != nil {
+			return nil, err
+		}
+
+		if s.metadata.RemoteMixerDomain != "" {
```

Comment:
maybe a TODO that these two calls can be done concurrently.

Reply:
Added a TODO in the top of the file, as this applies to all V2 APIs.